contrib/epee/include/net/net_helper.h

```diff
@@ -146,38 +153,61 @@ namespace net_utils
 				//boost::asio::ip::tcp::endpoint remote_endpoint(boost::asio::ip::address::from_string(addr.c_str()), port);
 				boost::asio::ip::tcp::endpoint remote_endpoint(*iterator);
 
-
-				m_socket.open(remote_endpoint.protocol());
-				if(bind_ip != "0.0.0.0" && bind_ip != "0" && bind_ip != "" )
-				{
-					boost::asio::ip::tcp::endpoint local_endpoint(boost::asio::ip::address::from_string(addr.c_str()), 0);
-					m_socket.bind(local_endpoint);
-				}
-
-				
 				m_deadline.expires_from_now(timeout);
 
 
 				boost::system::error_code ec = boost::asio::error::would_block;
-
-				//m_socket.connect(remote_endpoint);
-				m_socket.async_connect(remote_endpoint, boost::lambda::var(ec) = boost::lambda::_1);
-				while (ec == boost::asio::error::would_block)
-				{	
-					m_io_service.run_one(); 
-				}
-				
-				if (!ec && m_socket.is_open())
+        
+				if(!m_ssl)
 				{
-					m_connected = true;
-					m_deadline.expires_at(std::chrono::steady_clock::time_point::max());
-					return true;
-				}else
+  				m_socket.open(remote_endpoint.protocol());
+  				if(bind_ip != "0.0.0.0" && bind_ip != "0" && bind_ip != "" )
+  				{
+  					boost::asio::ip::tcp::endpoint local_endpoint(boost::asio::ip::address::from_string(addr.c_str()), 0);
+  					m_socket.bind(local_endpoint);
+  				}
+
+  				//m_socket.connect(remote_endpoint);
+  				m_socket.async_connect(remote_endpoint, boost::lambda::var(ec) = boost::lambda::_1);
+  				while (ec == boost::asio::error::would_block)
+  				{
+  					m_io_service.run_one(); 
+  				}
+				
+  				if (!ec && m_socket.is_open())
+  				{
+  					m_connected = true;
+  					m_deadline.expires_at(std::chrono::steady_clock::time_point::max());
+  					return true;
+  				} else
+  				{
+  					MWARNING("Some problems at connect, message: " << ec.message());
+  					return false;
+  				}               
+				} 
+				// SSL
+				else
 				{
-					MWARNING("Some problems at connect, message: " << ec.message());
-					return false;
+  				boost::asio::async_connect(m_ssl_socket.lowest_layer(), iterator, boost::lambda::var(ec) = boost::lambda::_1);
+          
+  				while (ec == boost::asio::error::would_block)
+  				{
+  					m_io_service.run_one(); 
+  				}
+				
+  				if (!ec)
+  				{
+						m_connected = true;
+						m_deadline.expires_at(std::chrono::steady_clock::time_point::max());
```

Comment:
check_deadline is only used during an operation. When operation is finished, it's set to never timeout.